### DISCUSSION TITLE
Excluding certain changed files from a commit

### DISCUSSION THREAD

**Discussion by MAS-6798** [community/community#77408] [Repositories] [answered]
### Select Topic Area

Question

### Body

Brand new to GitHub, so apologies if there's a blindingly obvious answer to this, but I can't seem to find it.

I'm just migrating from TFS 2015 and something I found useful during a check in (TFS terminology) was to be able exclude certain files from the check in, so they'd remain checked out locally. It was as easy as moving files between "Included Changes" and "Excluded Changes".

I'd like something similar with Git in Visual Studio - when doing a commit to be able to exclude certain files that I may have changed but don't want to commit.

.gitignore doesn't seem to do the job because it's a more permanent way to exclude files (I think).

Is this possible? Thanks.

**Comment by b8zeek** [comment, marked as answer]
Greetings friend. 👋🏼 

First of all, good luck on your git journey! 🚀 

To answer the question, yes, of course, it's possible to exclude certain files from a commit in Git.

How can you do it? Follow the steps below:

Fist, you want to stage the changes you want to commit. You can do this using the `git add` command followed by the file or directory name e.g. `git add file1.txt`. You can also add all changes with `git add .`. In your case, if you have changes in some files that you don't want to commit, simply don't stage them. Git will only commit changes that have been staged.

Once you've staged all the changes you want to commit, you can commit them using `git commit -m "YOUR COMMIT MESSAGE."`.

The last step is to push the changes to the cloud with `git push`.

Visual Studio Code flow, you can also do this visually:

Open the Source Control view (the icon that looks like a branch in the left sidebar).

You'll see a list of changed files. You can stage a file by clicking the `+` icon next to its name. Files that have been staged will move to the "Staged Changes" section. Only these files will be included in the commit.

You can commit the staged changes by entering a commit message in the text box at the top and pressing `Commit` button or by `Ctrl` + `Enter`.

P.S. I would suggest using a Git GUI, my favourites are SourceTree, GitKraken and GitHub Desktop. Every one has its own advantages; it will be much easier for you to understand what is happening with the graphical representation.

I hopes this clarifies if for you. Cheers! 🍺

🍀

**Comment by MAS-6798** [comment]
Perfect - thanks for your answer, works exactly how I would like!

I've got GitHub Desktop, but will check out the others.

**Comment by b8zeek** [comment]
Great bro! 🚀 Feel free to tag me in case you need more assistance with this. 🍺

**SourceTree** is really clean, simple, understandable and offers such an enjoyable experience. I would say it is the best client if you work on GitHub, GitLab, Bitbucket... The most readable one, definitely! Beginner friendly.

**GitKraken** is GitHub-specialized and offers everything that you need in your workflow. It has client, IDE extension, browser extension, CLI... And it is smooth! It gives you the option to have the workspaces, too. It is better than the **GitHub Desktop** for most of the GitHub stuff. Power users client, definitely!

**GitHub Desktop** is the best for searching through the repos, cloning them, publishing the newly created ones... It is simple, for simple usage the best option as it doesn't offer many advanced features. Use it for your projects, where you're working alone. Beginner friendly!

This is a quick short summary. I use all three, depending on what I want to use them for, depending on how I feel at that moment or simply which UI I prefer at that time.

Best of luck. 🍺

🍀

**Reply by MAS-6798** [reply]
Just done now - you can see I also need help with good etiquette!

Got SourceTree now too, will take a look at GitKraken.

Thanks again - it's really refreshing to get such a good response to what's probably a very basic question. Enjoying the journey so far!

**Reply by b8zeek** [reply]
Love to hear that! 🚀 Keep it enjoyable. 🙏🏼 

🍀
